### PR TITLE
Add code analysis properties to project XSD

### PR DESCRIPTION
**Pull request by drewnoakes**
Fixes https://github.com/dotnet/project-system/issues/6600

Adds:
- `AnalysisMode`
- `EnforceCodeStyleInBuild`

### CODE REVIEW

File: src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 1]
```diff
@@ -1158,6 +1158,18 @@ elementFormDefault="qualified">
     <xs:element name="MinimumVisualStudioVersion" type="msb:StringPropertyType" substitutionGroup="msb:Property"/>
     <xs:element name="AdditionalFileItemNames" type="msb:StringPropertyType" substitutionGroup="msb:Property"/>
     <xs:element name="AllowUnsafeBlocks" type="msb:StringPropertyType" substitutionGroup="msb:Property"/>
+    <xs:element name="AnalysisMode" type="msb:StringPropertyType" substitutionGroup="msb:Property">
```
Comment (mavasani) [inline src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd:1161]: Thanks @drewnoakes. While you are adding code analysis related properties, can you also add 3 other boolean properties mentioned here: https://docs.microsoft.com/visualstudio/code-quality/disable-code-analysis?view=vs-2019#net-framework-projects? They apply to all .NET projects.

```xml
<RunAnalyzersDuringBuild>false</RunAnalyzersDuringBuild>
<RunAnalyzersDuringLiveAnalysis>false</RunAnalyzersDuringLiveAnalysis>
<RunAnalyzers>false</RunAnalyzers>
```

**Review by mavasani — APPROVED** [review]

File: src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd (hunk 1 above)
Comment (drewnoakes) [reply src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd]: @mavasani done.

**Comment by drewnoakes** [comment]
Fixed the previous issue (specifying both type and complex type on the same element) and added some more properties based on @mavasani's feedback.

**Review by rainersigwald — APPROVED** [review]

File: src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd (hunk 1 above)
Comment (mavasani) [reply src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd]: Thanks @drewnoakes!

**Review by mavasani — APPROVED** [review]

**Comment by Forgind** [comment]
Thanks!

**Pull request merged**